lib/src/modules/conditional.dart

```diff
@@ -117,13 +117,16 @@ class Combinational extends _Always {
 
     _isExecuting = true;
 
-    // combinational must always drive all outputs or else you get X!
-    for (var element in _assignedReceiverToOutputMap.values) {
-      element.put(LogicValue.x, fill: true);
+    var drivenLogics = <Logic>{};
+    for (var element in conditionals) {
+      drivenLogics.addAll(element.execute());
```

Comment:
Perhaps use some sort of counter/flag instead of constructing and combining sets would be better here.

Reply:
this is tricky to implement in a way that actually yields good performance boosts since each signal could be driven multiple times per `execute` and each `execute` could be called multiple times per timestamp.  the common case is that all items will be driven every cycle (in a "good" design), so the cost of iterating through all the flags would still be recurring.

this can be revisited if it turns out that the creation of the `Set`s is expensive enough to warrant a change